### PR TITLE
Add searchbar to docs

### LINKED ISSUE #709: Enable search on docs
https://docusaurus.io/docs/search

### PR DESCRIPTION
**Pull request by Etesam913**
### Login to algolia here: 
https://www.algolia.com/users/sign_in

### You can see search plugin here:
https://www.algolia.com/dashboard

### To trigger a manual crawl(website gets automatically crawled once every week) of the website login here:
https://crawler.algolia.com/

manual_crawl

_The button to click for a manual crawl_

## Double Search Result Issue (RESOLVED)
* Fixed issue of double search results by adding the below line in the crawler config(click editor after visiting crawler link above)
```exclusionPatterns: ["https://mephisto.ai//*/**"],```
* For some reason the crawler was picking up links with a double //, so I am just excluding them now
  * These were the search result links that were not working, so all the search result links should work now.

## Video

https://user-images.githubusercontent.com/55665282/180869527-22dcaa86-4d3f-4e0e-ace1-d3953c6d5b71.mov

### Issue to be resolved later:
* Clicking the "see all results" text takes you to a page that allows you to "see all results."
  * The results on this page do not load.
    * This has to do with the contextualSearch property being enabled on this page.
    * With it set to false, search works in the modal view, but fails in the "see all results" page.
  * With contextualSearch set to true, the search fails everywhere
  * While I set contextualSearch to false in my properties it doesn't get applied on this search more page.
    * I tried looking at this link, did not seem to help me much: https://github.com/facebook/docusaurus/issues/3805
    * I also tried fixing the issue with contextualSearch itself, but that does not seem to work. (https://github.com/facebook/docusaurus/issues/6693)
  * **I just removed the "see more results" text with CSS so the user won't go to this page.**

## Video of the bug (I unhid the "see all results" text for this video) 

https://user-images.githubusercontent.com/55665282/180871282-3878ce51-02df-489b-bdd2-74fde81cddae.mov

### CODE REVIEW

File: docs/web/docusaurus.config.js [hunk 1]
```diff
@@ -42,6 +42,29 @@ const config = {
...
+        // Public API key: it is safe to commit it
+        apiKey: "ac51a5b25842fad8a3a7b1f384496bf9",
+
+        indexName: "docsearch",
+
+        // Optional: see doc section below
+        contextualSearch: true,
+
+        // Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.
+        externalUrlRegex: "external\\.com|domain\\.com",
```
Comment (pringshia) [inline docs/web/docusaurus.config.js:58]: is this correct?

**Review by pringshia — APPROVED** [review]
Thanks for adding this! It immediately makes the docs way more useful and accessible.

Have one small comment re: a config value, but otherwise LGTM.

File: docs/web/docusaurus.config.js [hunk 2]
```diff
@@ -42,6 +42,29 @@ const config = {
...
     /** @type {import('@docusaurus/preset-classic').ThemeConfig} */
     ({
+      algolia: {
+        // The application ID provided by Algolia
+        appId: "J6ARWT70HK",
+
+        // Public API key: it is safe to commit it
+        apiKey: "ac51a5b25842fad8a3a7b1f384496bf9",
+
+        indexName: "docsearch",
```
Comment (pringshia) [inline docs/web/docusaurus.config.js:52]: change indexName to "mephisto"

**Review by pringshia — CHANGES_REQUESTED** [review]
actually... some comments need to be addressed first

File: docs/web/docusaurus.config.js [hunk 3]
```diff
@@ -50,20 +50,6 @@ const config = {
...
-        contextualSearch: true,
-
-        // Optional: Specify domains where the navigation should occur through window.location instead on history.push. Useful when our Algolia config crawls multiple documentation sites and we want to navigate with window.location.href to them.
-        externalUrlRegex: "external\\.com|domain\\.com",
-
-        // Optional: Algolia search parameters
-        searchParameters: {},
-
-        // Optional: path for search page that enabled by default (`false` to disable it)
-        searchPagePath: "search",
```
[the remark this answers is not in the document]
Comment (Etesam913) [reply docs/web/docusaurus.config.js]: I set searchPagePath to false

**Review by pringshia — APPROVED** [review]
ty!

**Pull request merged**